core/server/worker/src/main/java/alluxio/worker/page/PagedBlockStore.java

```diff
@@ -507,4 +510,33 @@ private int getDirIndexOfBlock(long blockId) {
         .getDir()
         .getDirIndex();
   }
+
+  private void removeBlockInternal(long sessionId, long blockId, long timeoutMs)
+      throws IOException {
+    Optional<BlockLock> optionalLock =
+        mLockManager.tryAcquireBlockLock(sessionId, blockId, BlockLockType.WRITE,
+        timeoutMs, TimeUnit.MILLISECONDS);
+    if (!optionalLock.isPresent()) {
+      throw new DeadlineExceededException("Cannot acquire lock to remove block");
+    }
+    try (BlockLock blockLock = optionalLock.get()) {
+      Set<PageId> pageIds;
+      try (LockResource metaLock = new LockResource(mPageMetaStore.getLock().writeLock())) {
+        if (mPageMetaStore.hasTempBlock(blockId)) {
+          throw new IllegalStateException(
+            ExceptionMessage.REMOVE_UNCOMMITTED_BLOCK.getMessage(blockId));
+        }
+        pageIds = mPageMetaStore.getBlock(blockId)
+          .orElseThrow(() -> new BlockDoesNotExistRuntimeException(blockId))
+          .getDir().getBlockPages(blockId);
+
+        for (PageId pageId : pageIds) {
+          PageInfo pageInfo = mPageMetaStore.removePage(pageId);
+          pageInfo.getLocalCacheDir().getPageStore().delete(pageId);
+        }
+      }
+    } catch (PageNotFoundException e) {
+      throw AlluxioRuntimeException.from(e);
```

Comment:
Conversion from a `PageNotFoundException` which is a `AlluxioException` to `AlluxioRuntimeException` is not well-defined (it gets converted to a `UnknownRuntimeException`. Consider manually constructing a `NotFoundRuntimeException` from the underlying exception.

Reply:
@dbw9580 Done. PTAL thanks!